src/modules.rs

```diff
@@ -91,51 +91,71 @@ impl Modules {
 pub struct Deps {
   pub name: String,
   pub deps: Option<Vec<Deps>>,
-  depth: usize,
+  prefix: String,
+  is_last: bool,
 }
 
 impl Deps {
   pub fn new(modules: &Modules, module_name: &str) -> Deps {
     let mut seen = HashSet::new();
     let id = modules.get_id(module_name).unwrap();
-    Self::helper(&mut seen, 0, modules, id)
+    Self::helper(&mut seen, "".to_string(), true, modules, id)
   }
 
   fn helper(
     seen: &mut HashSet<deno_mod>,
-    depth: usize,
+    prefix: String,
+    is_last: bool,
     modules: &Modules,
     id: deno_mod,
   ) -> Deps {
     let name = modules.get_name(id).unwrap().to_string();
     if seen.contains(&id) {
       Deps {
         name,
-        depth,
+        prefix,
         deps: None,
+        is_last,
       }
     } else {
       seen.insert(id);
       let child_ids = modules.get_children(id).unwrap();
+      let child_count = child_ids.iter().count();
       let deps = child_ids
         .iter()
-        .map(|dep_id| Self::helper(seen, depth + 1, modules, *dep_id))
-        .collect();
+        .enumerate()
+        .map(|(index, dep_id)| {
+          let new_is_last = index == child_count - 1;
+          let mut new_prefix = prefix.clone();
+          new_prefix.push(if is_last { ' ' } else { '│' });
+          new_prefix.push(' ');
+          Self::helper(seen, new_prefix, new_is_last, modules, *dep_id)
+        }).collect();
       Deps {
         name,
-        depth,
+        prefix,
         deps: Some(deps),
+        is_last,
       }
     }
   }
 }
 
 impl fmt::Display for Deps {
   fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-    for _i in 0..self.depth {
-      write!(f, "| ")?;
+    let mut has_children = false;
+    if let Some(ref deps) = self.deps {
```

Comment:
not sure if there's a more straightforward way of doing basically `obj.array && obj.array.length > 0` here

Reply:
What you have is good.